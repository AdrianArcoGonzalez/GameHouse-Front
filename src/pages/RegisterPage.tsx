import heroTexts from "../components/HeroSection/heroData/heroData";
import HeroSection from "../components/HeroSection/HeroSection";
import Register from "../components/Register/Register/Register";

const RegisterPage = (): JSX.Element => {
  return (
    <>
      <HeroSection
        text={heroTexts.register}
        srcImage={"images/register.webp"}
      />
      <Register />;
    </>
  );
};

export default RegisterPage;
