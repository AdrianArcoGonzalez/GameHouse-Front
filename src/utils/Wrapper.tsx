import * as reactRedux from "react-redux";
import { store } from "../store/store";

export interface WrapperProps {
  children: JSX.Element | JSX.Element[];
}

export const Wrapper = ({ children }: WrapperProps): JSX.Element => {
  return <reactRedux.Provider store={store}>{children}</reactRedux.Provider>;
};
