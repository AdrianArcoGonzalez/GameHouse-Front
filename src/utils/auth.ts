import jwt from "jwt-decode";

const decodeToken = (token: string) => {
  const payloadToken: {
    id: string;
    username: string;
    iat: number;
  } = jwt(token);

  const user = {
    token: token,
    id: payloadToken.id,
    username: payloadToken.username,
    isLogged: true,
  };
  return user;
};

export default decodeToken;
