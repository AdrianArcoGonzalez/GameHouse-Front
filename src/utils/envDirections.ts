export const backUrl = process.env.REACT_APP_URL_BACK

export const urlsBack = {
  default: `${backUrl}games/games/`,
  getByOwner: `${backUrl}games/games/my-collection/`,
  getByCategory: `${backUrl}games/games/category/`,

}
