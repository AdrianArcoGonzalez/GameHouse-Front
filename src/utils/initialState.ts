import decodeToken from "./auth";

const emptyUser = {
  id: "",
  username: "",
  token: "",
  isLogged: false,
};
const initialState = () => {
  const token = localStorage.getItem("token") as string;

  if (token) {
    return decodeToken(token);
  }

  return emptyUser;
};

export default initialState;
