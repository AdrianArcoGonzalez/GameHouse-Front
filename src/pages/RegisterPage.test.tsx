import TestRenderer from "react-test-renderer";
import RegisterPage from "./RegisterPage";
import { Wrapper } from "../utils/Wrapper";
import "react-toastify/dist/ReactToastify.css";

describe("Given a Register Page component", () => {
  describe("When it's instantiated", () => {
    test("Then it should match the Register page snapshot", () => {
      const expectedRegisterPage = TestRenderer.create(
        <Wrapper>
          <RegisterPage />
        </Wrapper>
      );

      expect(expectedRegisterPage).toMatchSnapshot();
    });
  });
});
