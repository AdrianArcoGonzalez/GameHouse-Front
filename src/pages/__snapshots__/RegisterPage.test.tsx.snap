// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Register Page component When it's instantiated Then it should render the Register page 1`] = `
Array [
  <section
    className="sc-bczRLJ DXVGo"
    data-testid="section"
  >
    <img
      alt="backgroundImage"
      className="hero__image"
      src="images/form-background.png"
    />
    <div
      className="hero-container"
    >
      <p
        className="hero-container__sentence"
      >
        Become a soldier of the GameHouse Army. Make your own reviews and receive feedback from other soldiers.
      </p>
    </div>
  </section>,
  <section
    className="sc-hKMtZM bMTXCn"
  >
    <h2
      className="register-title"
    >
      Sign Up
    </h2>
    <div
      className="register-container"
    >
      <div
        className="sc-gsnTZi gKoTxJ"
        data-testid="cardRegister"
      >
        <div
          className="card"
        >
          <span
            className="card__title"
          >
            GameHouse
          </span>
          <div
            className="card__user"
          >
            <ul
              className="card__left-side"
            >
              <li
                className="left-side__info-element"
              />
              <li
                className="left-side__info-element"
              />
              <li
                className="left-side__info-element"
              />
              <li
                className="left-side__info-element"
              />
            </ul>
            <div
              className="card__right-side"
            >
              <img
                alt="avatar"
                className="card__right-side--image"
                height="100px"
                src="https://aui.atlassian.com/aui/latest/docs/images/avatar-person.svg"
                width="100px"
              />
              <span
                className="card__right-side--username"
              />
            </div>
          </div>
        </div>
      </div>
      <form
        autoComplete="off"
        className="sc-dkzDqf iAGa-Di"
        data-testid="formRegister"
        noValidate={true}
        onSubmit={[Function]}
      >
        <label
          className="form__input-container"
          htmlFor="name"
        >
          Name
        </label>
        <input
          className="form__input-element"
          id="name"
          onChange={[Function]}
          placeholder="name"
          type="text"
          value=""
        />
        <label
          className="form__input-container"
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="form__input-element"
          id="email"
          onChange={[Function]}
          placeholder="john@gmail.com"
          type="email"
          value=""
        />
        <label
          className="form__input-container"
          htmlFor="birthdate"
        >
          Birthdate
        </label>
        <input
          className="form__input-element"
          id="birthdate"
          onChange={[Function]}
          placeholder="20/02/2020"
          type="date"
          value=""
        />
        <label
          className="form__input-container"
          htmlFor="location"
        >
          Location
        </label>
        <input
          className="form__input-element"
          id="location"
          onChange={[Function]}
          placeholder="Barcelona"
          type="text"
          value=""
        />
        <label
          className="form__input-container"
          htmlFor="image"
        >
          Image
        </label>
        <input
          className="form__input-element"
          id="image"
          onChange={[Function]}
          type="file"
          value=""
        />
        <label
          className="form__input-container"
          htmlFor="username"
        >
          Username
        </label>
        <input
          className="form__input-element"
          id="username"
          onChange={[Function]}
          placeholder="Gamer123"
          type="text"
          value=""
        />
        <label
          className="form__input-container"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form__input-element"
          id="password"
          onChange={[Function]}
          placeholder="********"
          type="password"
          value=""
        />
        <label
          className="form__input-container"
          htmlFor="repeatPassword"
        >
          Repeat Password
        </label>
        <input
          className="form__input-element"
          id="repeatPassword"
          onChange={[Function]}
          placeholder="********"
          type="password"
          value=""
        />
        <button
          className="form-button"
          disabled={true}
          type="submit"
        >
          Register
        </button>
      </form>
    </div>
  </section>,
  ";",
]
`;
